### ISSUE TITLE
Fluent: Cannot create VNET with existing route table attached by ID

### ISSUE THREAD

**Issue by jefflill** [Azure/azure-sdk-for-net#3175]
I'm trying to workaround #3171 by retrieving the ID of the route table I just created and then using the ID to attach the table to a new VNET.  This fails with:

&nbsp;&nbsp;&nbsp;&nbsp;`CloudException("Cannot parse the request")`

Here's the code:

````
using System;

using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;

namespace AzureMultpleRoutes
{
    class Program
    {
        static void Main(string[] args)
        {
            // NOTE: You'll need to replace these with your own Azure credentials.

            const string clientId       = "****";
            const string clientSecret   = "****";
            const string tenantId       = "****";
            const string subscriptionId = "****";

            const string resourceGroup  = "azure-bug";
            const string region         = "westus2";

            // Login the API.

            var azureCredentials =
                new AzureCredentials(
                    new ServicePrincipalLoginInformation()
                    {
                        ClientId     = clientId,
                        ClientSecret = clientSecret
                    },
                tenantId,
                AzureEnvironment.AzureGlobalCloud);

            var azure = Azure.Configure()
                .Authenticate(azureCredentials)
                .WithSubscription(subscriptionId);

            // Create the resource group if necessary.

            if (!azure.ResourceGroups.CheckExistence(resourceGroup))
            {
                azure.ResourceGroups
                    .Define(resourceGroup)
                    .WithRegion(region)
                    .Create();
            }

            // Create a routing table with two routes.
            //
            // Note that this throws:
            //
            //      ArgumentException("An item with the same key has already been added. Key: routes").
            //
            // but the route table was actually created correctly.

            string routeTableId;

            try
            {
                var routeTable = azure.Networks.Manager.RouteTables
                    .Define("routes")
                    .WithRegion(region)
                    .WithExistingResourceGroup(resourceGroup)
                    .DefineRoute("route-1")
                        .WithDestinationAddressPrefix("10.0.0.0/24")
                        .WithNextHopToVirtualAppliance("10.1.0.4")
                        .Attach()
                    .DefineRoute("route-2")
                        .WithDestinationAddressPrefix("10.0.1.0/24")
                        .WithNextHopToVirtualAppliance("10.1.0.5")
                        .Attach()
                    .Create();

                routeTableId = routeTable.Id;
            }
            catch (ArgumentException)
            {
                using (var networkClient = new Microsoft.Azure.Management.Network.NetworkManagementClient(azureCredentials))
                {
                    networkClient.SubscriptionId = subscriptionId;
                    routeTableId = networkClient.RouteTables.GetWithHttpMessagesAsync(resourceGroup, "routes").Result.Body.Id;
                }
            }

            azure.Networks
                .Define("network")
                .WithRegion(region)
                .WithExistingResourceGroup(resourceGroup)
                .WithAddressSpace("10.0.0.0/24")
                .DefineSubnet("subnet")
                    .WithAddressPrefix("10.0.0.0/24")
                    .WithExistingRouteTable(routeTableId)
                    .Attach()
                .Create();      // <--- Throws: CloudException("Cannot parse the request")
        }
    }
}
````
Note that this still fails even if I remove the second route definition so the route table creation succeeds and then use the ID of the route object returned.  Here's a more simple example:
````
using System;

using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;

namespace AzureMultpleRoutes
{
    class Program
    {
        static void Main(string[] args)
        {
            // NOTE: You'll need to replace these with your own Azure credentials.

            const string clientId       = "****";
            const string clientSecret   = "****";
            const string tenantId       = "****";
            const string subscriptionId = "****";

            const string resourceGroup  = "azure-bug";
            const string region         = "westus2";

            // Login the API.

            var azureCredentials =
                new AzureCredentials(
                    new ServicePrincipalLoginInformation()
                    {
                        ClientId     = clientId,
                        ClientSecret = clientSecret
                    },
                tenantId,
                AzureEnvironment.AzureGlobalCloud);

            var azure = Azure.Configure()
                .Authenticate(azureCredentials)
                .WithSubscription(subscriptionId);

            // Create the resource group if necessary.

            if (!azure.ResourceGroups.CheckExistence(resourceGroup))
            {
                azure.ResourceGroups
                    .Define(resourceGroup)
                    .WithRegion(region)
                    .Create();
            }

            var routeTable = azure.Networks.Manager.RouteTables
                .Define("routes")
                .WithRegion(region)
                .WithExistingResourceGroup(resourceGroup)
                .DefineRoute("route-1")
                    .WithDestinationAddressPrefix("10.0.0.0/24")
                    .WithNextHopToVirtualAppliance("10.1.0.4")
                    .Attach()
                .Create();

            azure.Networks
                .Define("network")
                .WithRegion(region)
                .WithExistingResourceGroup(resourceGroup)
                .WithAddressSpace("10.0.0.0/24")
                .DefineSubnet("subnet")
                    .WithAddressPrefix("10.0.0.0/24")
                    .WithExistingRouteTable(routeTable.Id)
                    .Attach()
                .Create();      // <--- Throws: CloudException("Cannot parse the request")
        }
    }
}
````

**Comment by jefflill** [comment]
One more thing: This workaround used to work with the old September 2016 beta release.

**Comment by jefflill** [comment]
**Correction:** This used to work with the Feb 2017 **1.0.0-beta50** release.

**Comment by martinsawicki** [comment]
after fixing https://github.com/Azure/azure-sdk-for-net/issues/3171, I don't repro this issue, so maybe there was some underlying problem with that still.
Our new [test case](https://github.com/Azure/azure-sdk-for-net/pull/3183/files#diff-1c0d5da5cc32ce9f1e54ef2b91781cf1) should cover this ok.

BTW - FYI - it is not needed to create a new authenticated client just to get to the underlying "auto-gen'd" implementations -- see the `.Inner` property on the fluent manager objects, "collection" objects and individual model objects. For example:
- `azure.Networks.Manager.Inner`
- `azure.Networks.Inner`
- `INetwork.Inner`
etc.

**Comment by jefflill** [comment]
Cool.  Thanks for the tip.

**Comment by hovsepm** [comment]
Fixed in v1.1.2. Please update your reference.
